invenio_administration/views/base.py

```diff
@@ -165,30 +159,47 @@ def _get_resource(cls, extension_name=None):
 
     @classmethod
     def _get_service_schema(cls):
+        """Get marshmallow schema of the assigned service."""
         # schema.schema due to the schema wrapper imposed,
         # when the actual class needed
         return cls.resource.service.schema.schema()
 
     def _schema_to_json(self, schema):
+        """Translate marshmallow schema to JSON to
+        provide action payload template for the frontend"""
         return jsonify_schema(schema)
 
-    def serialize_actions(self):
-        """Serialize actions for the resource view.
-
-        {"action_name":
-        {"text": "BLA"
-         #"api_endpoint": either from links.actions or provided by the resource
+    def get_api_endpoint(self):
+        """Get search API endpoint."""
+        api_url_prefix = current_app.config["SITE_API_URL"]
+        slash_tpl = "/" if not self.api_endpoint.startswith("/") else ""
 
-         If provided by resource,
-         there will be no mapping on the frontend side.
+        if not self.api_endpoint.startswith(api_url_prefix):
+            return f"{api_url_prefix}{slash_tpl}{self.api_endpoint}"
 
-         Also means we don't have links generated with permissions.
-         Best to have a specialized component,
-         which knows from where to read the endpoint.
+        return f"{slash_tpl}{self.api_endpoint}"
 
-         "payload_schema": schema in json
+    def serialize_actions(self):
+        """Serialize actions for the resource frontend view.
 
+        {"action_name":
+            {"text": "BLA"
+             "payload_schema": schema in json
+             "order": 1
+             }
+         }
         """
+        serialized_actions = {}
+        for key, value in self.actions:
+            if "payload_schema" and "order" not in value:
+                raise InvalidActionsConfiguration
+
+            if value["payload_schema"] is not None:
```

Comment:
`value["payload_schema"]` was already validated before, I think it's safe to access here